server/libs/common/src/utils/time-utils.ts

```diff
@@ -19,30 +19,33 @@ function createTimeUtils() {
 
   const getTimestampFromExif = async (originalPath: string): Promise<string> => {
     try {
-      const exifData = await exifr.parse(originalPath, {
-        tiff: true,
-        ifd0: true as any,
-        ifd1: true,
-        exif: true,
-        gps: true,
-        interop: true,
-        xmp: true,
-        icc: true,
-        iptc: true,
-        jfif: true,
-        ihdr: true,
-      });
+      const exifData = await exiftool.read(originalPath);
 
       if (exifData && exifData['DateTimeOriginal']) {
-        return exifData['DateTimeOriginal'];
+        await exiftool.end();
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return exifData['DateTimeOriginal'].toString()!;
       } else {
         return new Date().toISOString();
       }
     } catch (error) {
       return new Date().toISOString();
     }
   };
-  return { checkValidTimestamp, getTimestampFromExif };
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const getTimestampFromFilename = async (originalPath: string): Promise<string> => {
```

Comment:
could we remove the `@ts-ignore` (and eslint-disable)? This function would return either a promise that resolves to a string or undefined.

Then we can check if it resolves with a timestamp string in `server/apps/microservices/src/processors/metadata-extraction.processor.ts`  like:
```typescript
     if (!exifData || ! timestampFromFilename) {
        throw new Error(`can not parse exif data from file ${asset.originalPath}`);
      }
```

We can also add a test scenario for when it returns undefined (hypothetical case were the filename has a different naming format that doesn't match the current regex) 😃

Reply:
That sounds like a good idea. However I'm not quite sure as to what kind of test case should be added for that hypothetical case. :D